modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/RegionClassLoader.java

```diff
@@ -273,7 +267,7 @@ public final URL findResource(final String name) {
         if (exportingArtifactClassLoaders != null) {
           for (ArtifactClassLoader artifactClassLoader : exportingArtifactClassLoaders) {
             BundleDescriptor descriptor = artifactClassLoader.getArtifactDescriptor().getBundleDescriptor();
-            if (isRequestedArtifact(descriptor, groupId, artifactId, version, () -> {
+            if (descriptor != null && isRequestedArtifact(descriptor, groupId, artifactId, version, () -> {
```

Comment:
This was added due to test runner scenarios, right? I would add a comment to this line